crates/wallet/src/lib.rs

```diff
@@ -64,20 +64,20 @@ pub async fn create_mint_quote(
     db_conn: &Connection,
     node_client: &mut NodeClient<Channel>,
     method: String,
-    amount: u64,
+    amount: Amount,
     unit: String,
 ) -> Result<MintQuoteResponse> {
     let response = node_client
         .mint_quote(MintQuoteRequest {
             method: method.clone(),
-            amount,
+            amount: amount.into(),
             unit: unit.clone(),
             description: None,
         })
         .await?
         .into_inner();
 
-    db::store_mint_quote(db_conn, method, amount, unit, &response)?;
+    db::store_mint_quote(db_conn, method, amount.into(), unit, &response)?;
```

Comment:
You can change the signature of `store_mint_quote` to take an Amount instead of `u64` too.